CHANGES.md

```diff
@@ -1,8 +1,461 @@
+* switched ISx's arrays to use anonymous domains for a performance boost
+* added a new 'regexdna-redux' benchmark
+* made a number of speed and clarity improvements to the 'fasta' benchmark
+* made readability and correctness improvements to our 'pidigits' benchmark
+* fixed an off-by-one error in our 'binarytrees' benchmark
+* added missing Makefiles and fixed support for 'make -j'
+
+Documentation
+-------------
+* added new users guide sections on promotion, constants, type aliases, configs
+  (see TODO)
+* revised QUICKSTART instructions for clarity
+  (see $CHPL_HOME/QUICKSTART.rst -- TODO: is this right?)
+* reorganized the doc/ directory
+  (see TODO)
+* removed $CHPL_HOME/STATUS in favor of GitHub issues
+* updated bugs.rst to refer users to GitHub issues
+* improved the Docker README information
+* reorganized the platform-specific documentation pages into categories
+  (see http://chapel.cray.com/docs/master/platforms/index.html)
+* added documentation for dim() and dims() on arrays
+  (see http://chapel.cray.com/docs/master/builtins/internal/ChapelArray.html?highlight=dims#ChapelArray.dims)
+* fixed the documentation for string.strip()
+  (see TODO)
+* updated documentation regarding reference counting of files and channels
+* added an indication that IO is a module that is used by default
+* updated documents to refer to install.html rather than download.html
+* improved some entries in the "quick reference" document
+  (see TODO)
+
+Compiler Flags (see 'man chpl' for details)
+-------------------------------------------
+* added a '--[no-]print-callgraph flag to print a Chapel program's callgraph
+* added a '--[no-]div-by-zero-checks' flag to disable divide-by-0 checks
+
+Locale Models
+-------------
+* added a locale model for KNL with support for different memory types
+  (see TODO)
+* for 'numa', arrays are now allocted using multiple per-numa-domain chunks
+
+Portability
+-----------
+* added Chapel support for AWS EC2
+  (see TODO)
+* verified that Chapel works with the Windows 10 bash shell
+  (no platform-specific notes required... works like any other *nix)
+* added support for multi-locale ARM executions
+  (see TODO)
+
+Cray-specific Changes
+---------------------
+* made Cray systems always default to CHPL_COMM=ugni
+* stopped limiting the number of ugni communication domains under slurm
+* tuned the amount of task spinning before sleeping for Crays
+* removed extraneous/incorrect -O flags from cray-prgenv-cray command lines
+* make the heap used by CHPL_COMM=ugni 'numa'-localized when appropriate
+* fixed a problem when using 'muxed' tasking with 'gasnet' but no hugepages
+* enabled support for CHPL_COMM=ugni for the pgi compiler
+* fixed a bug in dynamic linking for gasnet-aries
+
+Platform-specific Changes
+-------------------------
+* fixed an occasional cygwin failure caused by pthread_attr_init()
+
+Error Messages
+--------------
+* improved const-checking in the compiler
```

Comment:
To clarify: This bullet has been here since Saturday night.  I wasn't asking about its positioning so much as whether you thought something more needed to be said.